src/render/painter.js

```diff
@@ -823,6 +823,7 @@ class Painter {
     prepareDrawProgram(context: Context, program: Program<*>, tileID: ?UnwrappedTileID) {
         const fog = this.style && this.style.fog;
         if (fog) {
+            const temporalOffset = (this.frameCounter / 1000.0) % 1;
```

Comment:
>Is worth coalescing fog and skybox's temporal offset? That would make it more difficult to define the uniform precisely when it's needed since characters in the codebase and the gl api call to set the uniform is probably the only cost involved.
>If we did check, maybe maybe style.fog || style.hasLayerType('sky') is an adequate check? But then we need to ensure skybox and fog don't re-define the same uniform

Might be ok just as is since it's only used in two locations, maybe we could compress painter.frameCounter to be painter.frameTemporalOffset? So that the same offset (counter / 1000) % 1 is applied.